v2/options.go

```diff
@@ -42,6 +67,21 @@ type ReadWriteOption interface {
 	readWriteOption()
 }
 
+// SplitReadWriteOptions splits the rwopts by type into ReadOption and WriteOption slices.
+func SplitReadWriteOptions(rwopts ...ReadWriteOption) ([]ReadOption, []WriteOption) {
```

Comment:
I'm wondering if these should go in an internal package - there's no good reason to export them, I think. Most users should just give vararg options.

Reply:
Move to internal causes cycle in package imports. dropped it in favour of duplicate code for now.